### PR TITLE
[stable/redis] Add loadBalancerSourceRanges to redis master and slave

### LINKED ISSUE #16644: [stable/redis] Add loadBalancerSourceRanges to master
Add `master.service.loadBalancerSourceRanges` option

**Additional context**
When `master.service.type` is set to `LoadBalancer` it will be accessible from all ip ranges 0.0.0.0/0, for security reason we should let user to specify loadBalancerSourceRanges.

### PR DESCRIPTION
**Pull request by pradithya**
#### Is this a new chart
No

#### What this PR does / why we need it:
Add loadBalancerSourceRanges to master and/or slave service when the service type is `LoadBalancer`

#### Which issue this PR fixes
  - fixes #16644

#### Special notes for your reviewer:
None

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/redis/values.yaml [hunk 1]
```diff
@@ -356,6 +357,7 @@ slave:
     annotations: {}
     labels: {}
     loadBalancerIP:
+    # loadBalancerSourceRanges: ["10.0.0.0/8"]
```
Comment (juan131) [inline stable/redis/values.yaml:360]: Please update the **values-production.yaml** too

File: stable/redis/templates/redis-master-svc.yaml [hunk 2]
```diff
@@ -19,6 +19,12 @@ spec:
   {{ if eq .Values.master.service.type "LoadBalancer" -}} {{ if .Values.master.service.loadBalancerIP -}}
   loadBalancerIP: {{ .Values.master.service.loadBalancerIP }}
   {{ end -}}
+  {{if .Values.master.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
+  {{ with .Values.master.service.loadBalancerSourceRanges }}
+{{ toYaml . | indent 4 }}
+  {{- end }}
+  {{- end }}
```
Comment (juan131) [inline stable/redis/templates/redis-master-svc.yaml]: Please apply the changes below:

```diff
- {{if .Values.master.service.loadBalancerSourceRanges }}
+ {{ if and (eq.Values.master.service.type  "LoadBalancer") .Values.master.service.loadBalancerSourceRanges }}
  loadBalancerSourceRanges:
- {{ with .Values.master.service.loadBalancerSourceRanges }}
+ {{- with .Values.master.service.loadBalancerSourceRanges }}
{{ toYaml . | nindent 4 }}
  {{- end }}
```

Apply the same suggestion to **redis-slave-svc.yaml** service too.

File: stable/redis/Chart.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: redis
-version: 9.3.1
+version: 9.3.2
```
Comment (juan131) [inline stable/redis/Chart.yaml]: Please bump minor version instead

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @pradithya 

Thanks so much for your contribution! Could you please take a look to my suggestions?

File: stable/redis/Chart.yaml (hunk 3 above)
Comment (pradithya) [reply stable/redis/Chart.yaml]: Updated in 22b6f2234b366fdcd708e92ff306afe064d73e97

File: stable/redis/templates/redis-master-svc.yaml (hunk 2 above)
Comment (pradithya) [reply stable/redis/templates/redis-master-svc.yaml]: Applied in 24b885d

File: stable/redis/values.yaml (hunk 1 above)
Comment (pradithya) [reply stable/redis/values.yaml:360]: Updated in 87e63e0805590f197e9d3458dffbaf3ce830c363

**Comment by pradithya** [comment]
@juan131 I have applied all suggestions

**Review by juan131 — APPROVED** [review]
/ok-to-test

**Comment by juan131** [comment]
Hi @pradithya 

I'm approving your changes! Thanks so much

Could you please rebase from master and update the PR? Once done, we can continue with the merging process

**Comment by pradithya** [comment]
@juan131 Thanks for reviewing, I've rebased against mater

**Review by juan131 — APPROVED** [review]
/lgtm

**Pull request merged**